HackIllinois/Controllers/FlowControllers/HILoginFlowController.swift

```diff
@@ -114,47 +114,97 @@ extension HILoginFlowController {
 
 // MARK: - Login Flow
 extension HILoginFlowController {
-    func populateUserData(loginMethod: HILoginMethod, token: String, sender: HIBaseViewController) {
-        HIUserService.getUser(by: token, with: loginMethod)
-        .onCompletion { result in
-            switch result {
-            case .success(let containedUser, _):
-                let userInfo = containedUser.data[0]
-                var user = HIUser(
-                    loginMethod: loginMethod,
-                    permissions: userInfo.roles.map { $0.permissions }.reduce(.guest, max),
-                    token: token,
-                    identifier: userInfo.info.email,
-                    isActive: true,
-                    id: userInfo.info.id,
-                    name: nil,
-                    dietaryRestrictions: nil
-                )
-
-                HIRegistrationService.getAttendee(by: token, with: loginMethod)
-                .onCompletion { result in
-                    switch result {
-                    case .success(let containedAttendee, _):
-                        let attendeeInfo = containedAttendee.data[0]
-                        let names = [attendeeInfo.firstName, attendeeInfo.lastName].compactMap { $0 } as [String]
-                        user.name = names.joined(separator: " ")
-                        user.dietaryRestrictions = attendeeInfo.diet
-
-                    case .failure:
-                        break
-                    }
-
+    func getUserTokenFromCode(code: String) {
+        let json: [String: Any] = ["code": code]
+        let jsonData = try? JSONSerialization.data(withJSONObject: json)
+        // create post request
+        let url = URL(string: "https://api.hackillinois.org/auth/code/github/?redirect_uri=https://test.hackillinois.org/auth/?isiOS=1")!
+        var request = URLRequest(url: url)
+        request.httpMethod = "POST"
+        // insert json data to the request
+        request.httpBody = jsonData
+        let task = URLSession.shared.dataTask(with: request) { data, _, error in
+            guard let data = data, error == nil else {
+                return
+            }
+            let responseJSON = try? JSONSerialization.jsonObject(with: data, options: [])
+            if let responseJSON = responseJSON as? [String: Any] {
+                if let token = responseJSON["token"] as? String {
+                    let tok = token.trimmingCharacters(in: .whitespacesAndNewlines)
                     DispatchQueue.main.async {
-                        NotificationCenter.default.post(name: .loginUser, object: nil, userInfo: ["user": user])
+                        self.populateUserData(loginMethod: .github, token: tok, sender: self.loginSelectionViewController)
```

Comment:
What was the rational here, I don’t remember this code well

Reply:
as per the current login flow, the jwt is obtained for the user and then used to get user and attendee information about the user before posting to the notification center